### ISSUE TITLE
Add "go-tls-lint" linter

### ISSUE THREAD

**Issue by leonklingele** [golangci/golangci-lint#4350]
### Your feature request related to a problem? Please describe.

An opinionated Go code linter for detecting TLS config issues.

```go
func configSettings() {
	a := tls.Config{
		Renegotiation: tls.RenegotiateFreelyAsClient, // want `WARNING: Renegotiation is not supported in TLS 1.3`
		MinVersion:    tls.VersionTLS12,              // want `ERROR: Go 1.18 and onward has good defaults for MinVersion, no need to set it`
		Time:          time.Now,                      // should not raise issue
	}
	a.Renegotiation = tls.RenegotiateFreelyAsClient // want `WARNING: Renegotiation is not supported in TLS 1.3`
	a.MinVersion = tls.VersionTLS12                 // want `ERROR: Go 1.18 and onward has good defaults for MinVersion, no need to set it`
	a.Time = time.Now                               // should not raise issue

	b := &tls.Config{
		Renegotiation: tls.RenegotiateFreelyAsClient, // want `WARNING: Renegotiation is not supported in TLS 1.3`
		MinVersion:    tls.VersionTLS12,              // want `ERROR: Go 1.18 and onward has good defaults for MinVersion, no need to set it`
		Time:          time.Now,                      // should not raise issue
	}
	b.Renegotiation = tls.RenegotiateFreelyAsClient // want `WARNING: Renegotiation is not supported in TLS 1.3`
	b.MinVersion = tls.VersionTLS12                 // want `ERROR: Go 1.18 and onward has good defaults for MinVersion, no need to set it`
	b.Time = time.Now                               // should not raise issue
}
```

https://github.com/Azure/go-tls-lint

### Describe the solution you'd like.

—

### Describe alternatives you've considered.

—

**Comment by silverwind** [comment]
This seems like to have some overlap with existing `gosec` linter, is it worth having?

**Comment by ldez** [comment]
There is overlap with gosec and this linter cannot be used as a lib.
